### ISSUE TITLE
terminate called after throwing an instance of 'Json::LogicError'

### ISSUE THREAD

**Issue by snajperro** [ethereum-mining/ethminer#1654]
● miner.service
   Loaded: loaded (/etc/systemd/system/miner.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2018-10-23 16:47:04 MSK; 1h 15min ago
  Process: 2169 ExecStop=/bin/kill -s QUIT $MAINPID (code=exited, status=0/SUCCESS)
  Process: 1991 ExecStart=/home/rig1/sh/ethereum.sh (code=exited, status=134)
 Main PID: 1991 (code=exited, status=134)

Oct 23 16:46:56 192 ethereum.sh[1991]:  m 16:46:56 ethminer Speed 146.71 Mh/s gpu0 24.45 gpu1 24.45 gpu2 24.45 gpu3 24.45 gpu4 24.45 gpu5 24.45 A108+4:R1 Time: 00:49
Oct 23 16:47:01 192 ethereum.sh[1991]:  i 16:47:01 stratum  Job: #2a8b58bf… eu1.ethermine.org [172.65.207.106:4444]
Oct 23 16:47:01 192 ethereum.sh[1991]:  X 16:47:01 stratum  Got invalid Json message :* Line 1, Column 1
Oct 23 16:47:01 192 ethereum.sh[1991]:   Syntax error: value, object or array expected.
Oct 23 16:47:01 192 ethereum.sh[1991]: terminate called after throwing an instance of 'Json::LogicError'
Oct 23 16:47:01 192 ethereum.sh[1991]:   what():  in Json::Value::find(key, end, found): requires objectValue or nullValue
Oct 23 16:47:01 192 ethereum.sh[1991]: /home/rig1/sh/ethereum.sh: line 4:  2013 Aborted                 (core dumped) /home/rig1/ethereum/ethminer -U stratum1+tcp://0x4d46ed5fc8d41ec5f96ab9e24f8e9dce32ee5958.rig1@eu1.ethermine.org:4444
Oct 23 16:47:01 192 systemd[1]: miner.service: Main process exited, code=exited, status=134/n/a
Oct 23 16:47:04 192 systemd[1]: miner.service: Unit entered failed state.
Oct 23 16:47:04 192 systemd[1]: miner.service: Failed with result 'exit-code'.

**Comment by snajperro** [comment]
same on two machines, two month ago it has worked on old version, 
1)when i have run it . it haven't  worked yet.
2)i have installed last version of the miner(ethminer 0.17.0-alpha.1), but it doesn't have any effect.

**Comment by lesjokolat** [comment]
Can you tweak your startup:

/home/rig1/ethereum/ethminer -U -P stratum1+tcp://0x4d46ed5fc8d41ec5f96ab9e24f8e9dce32ee5958.rig1@eu1.ethermine.org:4444 -v 3

the extra verbosity will help troubleshoot.

**Comment by snajperro** [comment]
> Can you tweak your startup:
> 
> /home/rig1/ethereum/ethminer -U -P stratum1+tcp://0x4d46ed5fc8d41ec5f96ab9e24f8e9dce32ee5958.rig1@eu1.ethermine.org:4444 -v 3
> 
> the extra verbosity will help troubleshoot.

i will write log in 60 min

**Comment by snajperro** [comment]
https://pastebin.com/J3Ei7Ycj

**Comment by AndreaLanfranchi** [comment]
@snajperro 

```
Oct 23 18:47:56 192 ethereum.sh[2675]:  i 18:47:56 stratum  Job: #3b6cb240… eu1.ethermine.org [172.65.207.106:4444]
Oct 23 18:47:56 192 ethereum.sh[2675]:  X 18:47:56 stratum  Got invalid Json message :* Line 1, Column 1
Oct 23 18:47:56 192 ethereum.sh[2675]:   Syntax error: value, object or array expected.
Oct 23 18:47:56 192 ethereum.sh[2675]:  i 18:47:56 stratum  ":"
```

Two problems here : 

1. Pool is sending a malformed json message or the message gets tampered during it's trip from pool to your miner. Not much we can do about this. You may try to use stratum+tls and connect to ethermine.org's port 5555. See if the problem disappears;
2. Ethminer is not handling the malformed message properly and this is an error to be investigated.

**Comment by AndreaLanfranchi** [comment]
Solved by means of PR #1656
